src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionBase.cs

```diff
@@ -35,6 +35,7 @@ static string GetOrAddCachedValue([NotNull] ref string? cache, HeaderDescriptor
                 {
                     cache = lastValue = descriptor.GetHeaderValue(value, encoding);
                 }
+                Debug.Assert(cache is not null);
```

Comment:
Another nice example where it'd be good if the connection between lastValue and cache could be recognized by the compiler.

That said, the assert is fine. Thanks.